web/modules/custom/joinup_notification/src/EventSubscriber/CommunityContentSubscriber.php

```diff
@@ -249,32 +257,55 @@ protected function generateArguments(EntityInterface $entity) {
 
     $arguments['@actor:full_name'] = $actor_first_name . ' ' . $actor_last_name;
     $arguments['@transition:motivation'] = $motivation;
+    $arguments['@entity:hasPublished:status'] = $this->hasPublished ? 'an update of the' : 'a new';
 
     // Add arguments related to the parent collection or solution.
     $parent = $this->relationManager->getParent($entity);
-    if (!empty($parent)) {
-      $arguments['@group:title'] = $parent->label();
-      $arguments['@group:bundle'] = $parent->bundle();
-
-      // If the role is not yet set, get it from the parent collection|solution.
-      if (empty($arguments['@actor:role'])) {
-        $membership = $this->membershipManager->getMembership($parent, $actor);
-        if (!empty($membership)) {
-          $role_names = array_map(function (OgRoleInterface $og_role) {
-            return $og_role->getName();
-          }, $membership->getRoles());
-
-          if (in_array('administrator', $role_names)) {
-            $arguments['@actor:role'] = t('Owner');
-          }
-          elseif (in_array('facilitator', $role_names)) {
-            $arguments['@actor:role'] = t('Facilitator');
-          }
+    if (empty($parent)) {
+      return $arguments;
+    }
```

Comment:
No just wrap any code that needs to be skipped in an if statement. Putting a `return` in the middle of a method is poor practice since it can be easily missed when this code is changed in the future. In well written code the return statement is at the end.